### PR TITLE
RESO-1180: Add remove user tags and tests for NamedUserTagRequest

### PR DESCRIPTION
**Pull request by alexknipfer**
- Adds `remove` tags to payload for named user tag request
- There were currently no tests for this project, however, I went ahead and added tests for the file I touched
- Payload matches to API docs here: https://docs.airship.com/api/ua/index.html#operation-api-named_users-tags-post

Once we're good with these changes, we'll obviously have to publish a new version to NPM and update the version within the backend repo

### CODE REVIEW

File: src/lib/nameduser/NamedUserTagRequest.ts [hunk 1]
```diff
@@ -26,7 +39,19 @@ export class NamedUserTagRequest implements IRequest {
...
+      audience: this.audience,
+    }
+
+    if (Object.keys(this.add).length) {
+      payload['add'] = this.add
+    }
+
+    if (Object.keys(this.remove).length) {
+      payload['remove'] = this.remove
+    }
```
Comment (alexknipfer) [inline src/lib/nameduser/NamedUserTagRequest.ts:52]: I didn't see anything in the documentation mentioning if you can send these as an empty object if no tags are added for `add` or `remove`, but we know it works when you omit them, so I'm only adding them to the final payload if tags have been added to one or both.

**Review by bkma — APPROVED** [review]

File: src/lib/nameduser/NamedUserTagRequest.ts (hunk 1 above)
Comment (chatter) [reply src/lib/nameduser/NamedUserTagRequest.ts:52]: It does seem weird that someone could send both add and remove, but at the same time, that's kind of on them if they do something that silly :)

**Review by chatter — APPROVED** [review]

File: src/lib/nameduser/NamedUserTagRequest.ts (hunk 1 above)
Comment (alexknipfer) [reply src/lib/nameduser/NamedUserTagRequest.ts:52]: I could possibly see a use case to where you have specific tags you want to add and different for remove all within the same request

Comment (chatter) [reply src/lib/nameduser/NamedUserTagRequest.ts:52]: Are they both POST calls? I assumed removal of tags was a `DELETE` but yeah looking at the docs, guess it's possible. I keep forgetting they don't treat TAGS as a resource, which is *extremely* annoying.

Comment (alexknipfer) [reply src/lib/nameduser/NamedUserTagRequest.ts:52]: Yeah, just a POST call to `/named_users/tags`, with either or both `remove` and `add` properties

**Review by armonp — APPROVED** [review]

**Pull request merged**